lib/Server.js

```diff
@@ -121,11 +121,15 @@ const schema = require("./options.json");
  * @callback ByPass
  * @param {Request} req
  * @param {Response} res
- * @param {ProxyConfigArray} proxyConfig
+ * @param {ProxyConfigArrayItem} proxyConfig
```

Comment:
I think something wrong here `ByPass` can't have `ByPass` inside

Reply:
I haven't actually run the code yet, but looked at line 2222 it seemed to just pass the whole `proxyConfig` as-is:
```js
const isByPassFuncDefined =
  typeof proxyConfig.bypass === "function";
const bypassUrl = isByPassFuncDefined
  ? await proxyConfig.bypass(req, res, proxyConfig)
  : null;
```

Reply:
Can you provide configuration where we have a problem with types?

Reply:
Sure~ I've put a minimial repro. in the original issue #4161:
```ts
import type { Configuration } from 'webpack-dev-server'
const devServerOptions: Configuration = {
  proxy: [
    {
      // TS2322: Type '{ context: never[]; target: string; }' is not assignable to type 'Options'.
      // Object literal may only specify known properties, and 'context' does not exist in type 'Options'.
      context: [],
      target : '',
    },
  ],
}
```
That code works when using the old `@types/webpack-dev-server`, but throws TS error when using `webpack-dev-server`'s new bundle types.  (And it actually works if I just ignores the TS type error)

Reply:
We have more errors - try https://github.com/webpack/webpack-dev-server/pull/4173

Reply:
#4173 works for my case, should I close this one?

Reply:
Yep, let's close, merged https://github.com/webpack/webpack-dev-server/pull/4173, anyway thanks for PR